jml/utils/string_functions.cc

```diff
@@ -240,5 +240,21 @@ trim(const string & other)
     return result;
 }
 
+std::string
+join(const std::vector<std::string> strings)
```

Comment:
This should take a reference, and ideally we wouldn't add another function in string_functions.cc (which we want to remove).  Use boost::string instead.